data/transactions/logic/opcodes.go

```diff
@@ -141,7 +141,7 @@ var OpSpecs = []OpSpec{
 	{0x02, "keccak256", opKeccak256, asmDefault, disDefault, oneBytes, oneBytes, 2, modeAny, costly(130)},
 	{0x03, "sha512_256", opSHA512_256, asmDefault, disDefault, oneBytes, oneBytes, 2, modeAny, costly(45)},
 
-	{0x04, "ed25519verify", opEd25519verify, asmDefault, disDefault, threeBytes, oneInt, 1, runModeSignature, costly(1900)},
+	{0x04, "ed25519verify", opEd25519verify, asmDefault, disDefault, threeBytes, oneInt, 1, modeAny, costly(1900)},
```

Comment:
this is not right, `ed25519verify` must be disallowed in TEAL v2-v4 and allowed only in v5. So instead of removal, a new entry with the same opcode but different version is needed - see `txn` as an example. Plus a test verifying it is off in v2-v4 stateless and is on in v5

Reply:
Good point -- made two versions with v2-v4 with `runModeSignature` and v5 with `modeAny`